### DISCUSSION TITLE
Fire TV Client VLC works, but becomes unusable due to growing delay

### DISCUSSION THREAD

**Discussion by robosmith** [GerryDazoo/Slinger#185] [Q&A] [answered]
We got the server running on Raspberry Pi 3B: YAY! and Big Thanks to Gerry's new version.
However, VLC (PC and Fire TV) seem to have a video buffer that constantly grows with time used.
When it first starts up, the Remote command delay runs 8-10 seconds, but it steadily increases over time to an unusable 2 minutes.
Have experimented with OBS Studio (Win10) and that STAYS at 8-10 seconds delay between Remote command and screen update.
Have tried the other clients mentioned, but can't get any others to work.
Would really like to get a Fire TV client that does not suffer Remote command delays.
Thanks for any suggestions.

**Comment by GerryDazoo** [comment, marked as answer]
You can configure VLC to minimize it's buffering. Are you using a Solo/Pro/ProHD slingbox?
MX PLayer works OK my Firestick. Bt there is still a 4 secondish amount of buffereing

**Reply by robosmith** [reply]
Using ProHD slingbox. My friend tried MX Player, but could not get it to work. I will try that. Thanks.

**Reply by robosmith** [reply]
I got MX Player to work on Fire TV and it behaves similar to VLC. Starts out taking 10s for Remote commands and quickly grows to 30+ seconds. Does anyone have suggestions on how to limit the video buffer size? Thanks.

**Reply by GerryDazoo** [reply]
I think you're running out of stream on the PI. The Solo/Pro requires poking into every packet in the video stream to decrypt some data. For the 350/500/m1 it just get a packet/send a packet loop . Run a "top" command to see if the sling server is pegged at 100%.

**Reply by robosmith** [reply]
Yeah, it looks like the Pi 3b is pegged at 100% when streaming video. :(

**Reply by robosmith** [reply]
Since OBS Studio is not affected by length of play time and stays at ~7s  command delay, the increase in delay does not seem to be caused by overloading the Pi 3b.

**Reply by GerryDazoo** [reply]
Well then. It must be something on the client side and nothing to do with the server. But if I were you I'd reduce the videoBandwidth until your client can keep up. 99.99% of the delay is caused by buffering on the client side. Many articles on the web on how to minimize buffer delays for VLC. I've been using FFPLayer with minimal buffering and the remote is quite usable,

**Reply by robosmith** [reply]
Ok, replaced the Pi 3b with Pi 4 and that is only loaded 30%. Works much better with Mx Player and the Remote command delay does not seem to increase much from ~6-7s. I guess that the video interruption from the Pi being overloaded causes the buffer to grow and there does not seem to be a way to limit that in Mx Player, other than preventing the interruption. Not sure why OBS Studio is more resistant to buffer growth, but it does grow too with the Pi 3b, under severe Internet overload conditions.
